third-party/realdds/include/realdds/dds-stream.h

```diff
@@ -25,12 +36,25 @@ class dds_stream : public dds_stream_base
 protected:
     dds_stream( std::string const & stream_name, std::string const & sensor_name );
 
-    // dds_stream_base
 public:
-    bool is_open() const override;
-    bool is_streaming() const override;
+    bool is_open() const override { return !! _reader; }
+    virtual void open( std::string const & topic_name, std::shared_ptr< dds_subscriber > const & );
+    virtual void close();
+
+    bool is_streaming() const override { return _is_streaming; }
+    typedef std::function< void( const topics::device::image & f) > on_data_available_callback;

```

Comment:
I think you should be able to change this to non-const, giving us the ability to move the data out of it to another vector rather than having to allocate and copy each time.

Reply:
I tried but removing the `const` but I get compilation errors on the `invoke` call.
Changing to shared_ptr works but I'm not sure this is what we want

Lets talk tomorrow about the dispatcher in general and we can also solve this

Reply:
`on_data_available_callback` parameter is rvalue.
`mutable` specifier solves the lambda compilation problem (dispatcher removed so no lambda anyway)